client/signup/main.jsx

```diff
@@ -270,9 +270,15 @@ class Signup extends React.Component {
 		if ( this.props.stepName !== prevProps.stepName ) {
 			this.maybeShowSitePreview();
 			this.preloadNextStep();
+			// `scrollToTop` here handles cases where the viewport may fall slightly below the top of the page when the next step is rendered
```

Comment:
Looks like this only happens on a few edge cases e.g. scrolling just a little from the top and then progressing:

![calypso-signup-no-scroll-regression](https://user-images.githubusercontent.com/1705499/124804633-30e4f600-df63-11eb-8e91-b898a1c66ff9.gif)

I haven't been able to narrow in on why this happens, but it looks like we approached it similarly in Gutenboarding as well - see [onboarding-block/edit](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/gutenboarding/onboarding-block/edit.tsx#L83-L95) (props to @razvanpapadopol for pointing to this block of code).